DeepSpeech.py

```diff
@@ -212,20 +202,25 @@
 # ==============
 
 def variable_on_cpu(name, shape, initializer):
-    r"""
+    r'''
     Next we concern ourselves with graph creation.
     However, before we do so we must introduce a utility function ``variable_on_cpu()``
     used to create a variable in CPU memory.
-    """
-    # Use the /cpu:0 device for scoped operations
-    with tf.device('/cpu:0'):
+    '''
+    # Use the /cpu:0 device on worker_device for scoped operations
+    if len(FLAGS.ps_hosts) == 0:
+        device = worker_device
```

Comment:
Take a second look at the use of `worker_device`. Its use is not about CPU. There is also a CPU one. To make the global variables work in combination with the 'new' app start, this was the most obvious way to do it. However: I could also move all of them into the main routine or revert back to non-main (aka global code) based approach.